### ISSUE TITLE
Remove broken Intersphinx cross-references to Qiskit release notes

### ISSUE THREAD

**Issue by frankharkins** [Qiskit/documentation#617]
We need a way to reliably transform links of the form:

```
/api/qiskit/legacy_release_notes.*?
/api/qiskit/release_notes.*?
```

to the correct places in our documentation. This might be difficult as we split those files up, so we'd probably need to search through the files to find the correct anchors.

This is for #522 (see that issue for a full list of broken links); another option is to remove these entries from the `objects.inv`, which would mean other projects can't reference them, but I'm not sure how often the release notes are referenced anyway?

**Comment by Eric-Arellano** [comment]
> another option is to remove these entries from the objects.inv, which would mean other projects can't reference them, but I'm not sure how often the release notes are referenced anyway

I lean towards that for simplicity. Otherwise this is going to be pretty complex: like you suggest, we could try mapping anchors so that we know which link lives where. 

But we wouldn't have a solution for when there is no anchor. We don't know where a link to `release_notes.html` and `legacy_release_notes.html` goes; I guess `release-notes/index`?

You could test the impact of this out by starting up a local server with `python -m http.server` in a folder with `objects.inv`, and then try building the docs for Provider, and Runtime. There are other projects beyond those 2 that use Intersphinx, but it can give some indication.

**Comment by frankharkins** [comment]
After investigating, neither `qiskit-ibm-runtime` nor `qiskit-ibm-provider` reference the release notes using intersphinx. We think it's unlikely any other projects do, so we're fine to remove them.

**Comment by frankharkins** [comment]
Fixed in [1c8fc2c](https://github.com/Qiskit/documentation/pull/522/commits/1c8fc2c467c7c1e97f27e7d632b493e9110c0a4a)
